OpenRA.Mods.Common/Traits/Modifiers/WithColoredOverlay.cs

```diff
@@ -45,10 +51,11 @@ IEnumerable<IRenderable> ModifiedRender(Actor self, WorldRenderer wr, IEnumerabl
 			{
 				yield return a;
 
-				if (!a.IsDecoration)
-					yield return a.WithPalette(wr.Palette(Info.Palette))
-						.WithZOffset(a.ZOffset + 1)
-						.AsDecoration();
+				if (!a.IsDecoration && a is IModifyableRenderable)
```

Comment:
not sure if we should start using `if (!a.IsDecoration && a is IModifyableRenderable ma)`

Reply:
We should yeah, will fix.